package/MDAnalysis/lib/util.py

```diff
@@ -212,6 +214,182 @@ def filename(name, ext=None, keep=False):
     return name if isstream(name) else str(name)
 
 
+def get_reader_for(filename, format=None):
```

Comment:
So I changed where we declare the registry of readers so this isn't really circular, the timeline of reader registry looks like:

 * `MDAnalysis.__init__` declares that `_READERS` will hold a dict of readers, but is currently empty
 * `lib.util` gets given a reference to this (currently empty) dict, so it knows where to look for its functions
 * `coordinates.base` gets given the reference and fills it as `Reader`s are declared

It does sort of break the idea of `lib` being free from dependency, I could move the `get_*_for` to `core/` instead.